packages/webapp/src/containers/SensorReadings/saga.js

```diff
@@ -43,157 +43,193 @@ import i18n from '../../locales/i18n';
-    startDate = parseInt(+startDate / 1000);
-
-    const start = parseInt(+new Date().setDate(new Date().getDate() - 4) / 1000);
+  try {
     yield put(bulkSensorReadingsLoading());
-    const apikey = import.meta.env.VITE_WEATHER_API_KEY;
-    let params = {
-      appid: apikey,
-      lang: lang,
-      units: measurement,
-      type: HOUR,
-      start: start,
-      end: predictedEndDate,
-    };
+    const {
+      predictedUnixDate,
+      predictedUnixTime,
+      historicalUnixDate,
+      historicalUnixTime,
+      currentUnixDate,
+      currentUnixTime,
+    } = getDates();
+    const formattedEndDate = moment(predictedUnixTime).format('MM-DD-YYYY');
 
     const header = getHeader(user_id, farm_id);
-    const endDate = moment(new Date().setDate(new Date().getDate() + 2)).format('MM-DD-YYYY');
     const postData = {
       farm_id,
       user_id,
       locationIds,
       readingTypes,
-      endDate: endDate,
+      endDate: formattedEndDate,
     };
-    const result = yield call(axios.post, sensorReadingsUrl(), postData, header);
-    const allSensorNames = result?.data?.sensorsPoints.map((s) => s.name);
-    const centerPoint = findCenter(result?.data?.sensorsPoints.map((s) => s?.point));
-    const activeReadingTypes = result?.data?.sensorsPoints[0].reading_type;
-    const currentDT = parseInt(+new Date() / 1000);
-
-    let lastUpdatedReadingsTime = {};
-    let xAxisLabel = {};
-    let isFound = false;
-    let predictedXAxisLabel = '';
-    let stationName = '';
-    let latestTemperatureReadings = {};
-    let selectedSensorName = '';
-    let ambientDataWithSensorsReadings = {};
-    let latestActualReadTimes = [];
-    let sensorReadingData = {};
-
-    for (let readingType of readingTypes) {
-      latestActualReadTimes = result?.data?.sensorReading[readingType]
-        .filter((cv) => (cv.value ? cv.value : cv.value === 0))
-        .map((cv) => new Date(cv.actual_read_time).valueOf() / 1000);
 
-      lastUpdatedReadingsTime[readingType] = getLastUpdatedTime(latestActualReadTimes);
+    const result = yield call(axios.post, sensorReadingsUrl(), postData, header);
+    const data = result?.data;
+    let sensorDataByLocationIds = {};
 
-      if (readingType === TEMPERATURE) {
-        params = {
-          ...params,
-          lat: centerPoint?.lat ?? lat,
-          lon: centerPoint?.lng ?? lng,
-        };
-        const openWeatherPromiseList = [];
-        for (const weatherURL of OPEN_WEATHER_API_URL_FOR_SENSORS) {
-          const openWeatherUrl = new URL(weatherURL);
-          for (const key in params) {
-            openWeatherUrl.searchParams.append(key, params[key]);
-          }
-          if (weatherURL === DAILY_FORECAST_API_URL) {
-            openWeatherUrl.searchParams.append('cnt', 1);
-          }
-          openWeatherPromiseList.push(call(axios.get, openWeatherUrl?.toString()));
-        }
+    for (let locationId of locationIds) {
```

Comment:
@kathyavini 

I did add this didn't I.... I'm pretty sure it was because this hits the controller function `getAllSensorReadingsByLocationIds() `and that 
a) returns an array so I would have to hard code `locationId=locationIds[0]` 
b) is set up to return more locations but yes does not currently. 

....But I also don't expect charts to work out of the box if more location_ids are supplied. I think the main reason I went this way was the explicitness of locationIds and avoiding locationId=locationIds[0]. What do you think?

Reply:
How about

`const [locationId] = locationIds`

That would be a little bit nicer than the locationIds[0] and give a clearer idea of the current data shape.

Reply:
Im going to leave it for now if thats cool.